scene/main/canvas_item.h

```diff
@@ -70,6 +70,7 @@ class CanvasItem : public Node {
 		CLIP_CHILDREN_DISABLED,
 		CLIP_CHILDREN_ONLY,
 		CLIP_CHILDREN_AND_DRAW,
+		CLIP_CHILDREN_CANVASGROUP,
```

Comment:
```suggestion
		CLIP_CHILDREN_CANVAS_GROUP,
```

I think this better aligns with existing naming conventions.

Also, we should probably deprecate the `CanvasGroup` node, similar to how we removed the `YSort` node in 4.0 after the introduction of the `CanvasItem.y_sort_enabled` property. In any case, we should update the `CanvasGroup` documentation to reference this new mode.

Reply:
> the constant could be called `CLIP_CHILDREN_GROUP` and make as much sense, while being less repetitive

Yeah, I think that'd be better, especially if `CanvasGroup` is to be deprecated. 

My logic for the 'CanvasGroup' name suggestion was:
- Users familiar with 'CanvasGroup' node would get the idea what this mode does from its name;
- Users unfamiliar with 'CanvasGroup' could look up [the docs](https://docs.godotengine.org/en/stable/classes/class_canvasgroup.html).

Without `CanvasGroup` node in the editor, this name references nothing and is basically just a nod to a deprecated feature. Even [internally](https://github.com/godotengine/godot/blob/d61cd9149aaee6195b83bc65ca5539a27d76d61f/scene/2d/canvas_group.cpp#L37), the rendering mode is called 'transparent' but not 'CanvasGroup'. 

So agreed, it's better to avoid the 'CanvasGroup' name for this `ClipChildrenMode`. And I like the `Group` option.